README.adoc

```diff
@@ -0,0 +1,180 @@
+
+Again depending on the block, this chapter could invoke a hierarchy of sections and subsections. For simpler blocks,
+it might consist of a single Jinja template file named `HardwareInterface.jinja2`.
+
+== Appendix: Jinja Templates
+
+=== Overview
+Jinja templates consist of text interspersed with Jinja control statements.
+For a good introduction to Jinja, see the online https://jinja.palletsprojects.com/en/2.10.x/[Jinja Documentation].
+
+When Scribble invokes a Jinja template, it passes a number of predefined variables and template functions to the template.
+These variables and functions are described below.
+
+==== Predefined Variables for Onboarding New Blocks
+When documenting new blocks, each of the three sections will be passed the following predefined variables.
+
+
+scope:: The primary element being described. Typically this will be
+        based on the object model schema for the block.
+
+devices:: If there are multiple instances of the block, then
+  `devices` contains all of them.
+   Typically, `scope` is used to describe what all the devices
+   have in common, while `devices` is used to iterate through
+   the instances describing the unique values of each.
+
+product_name:: Generally, the new device will be part of a larger
+  design. `product_name` denotes the name of the larger design.
+  It is used mainly for titles, but it can be used by device sections
+  to remind the reader of the overall context around the device.
+
+
+=== Creating New Variables in Templates
+When invoking a template through the "Section()" call,
+all named parameters automatically become Jinja template variables.
+
+Thus, the following call to Jinja template "MyNewSection.jinja2"
+```
+Section(".MyNewSection", scope, name="Joe", **context)
+```
+will invoke the template "MyNewSection.jinja2",
+passing it the variable "name" with the value of "Joe".
+Note the section name `.MyNewSection` starts with a leading ".",
+so the file `MyNewSection.jinja2` should be in the same directory as the caller.
+
+
+==== Helper Functions for Displaying Device Data
+In addition to defining variables, Scribble defines a number of helper functions:
+
+base_addr(device):: Returns the hex memory address of a device.
+
+Figure(image_path, title="title", id=reference_id, width="50%")::
+Inserts a figure into the document. If the image_path starts with `{here}`,
+then the image file is in the same directory as the caller.
+Currently supports svg and png.
+
+RegisterMap(device):: Creates a register map of the device which can be used
+  to display various tables.  Typically,
+
+  {% set registers = RegisterMap(device) %}
+  {{ registers.table() }}            {# Displays a register map table #}
+  {{ registers.fields("ODATA") }}    {# Displays fields for register ODATA #}
+  {{ registers.names | human_list }} {# Displays a list of register names #}
+
+
+==== Jinja "Pipe" Expressions
+And finally, Scribble provides Jinja "pipe" expressions to help with
```

Comment:
We should use the terminology that Jinja uses for these things, which is "[filters](https://jinja.palletsprojects.com/en/2.10.x/templates/#filters)". Introducing our own terminology here is unnecessary.